File: src/main/java/com/jb/Project_coupon_3/models/Company.java
package com.jb.Project_coupon_3.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Table(name = "companies")
@Data
@NoArgsConstructor
public class Company {
    @Id
    @Setter(AccessLevel.NONE)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(nullable = false)
    @Setter(AccessLevel.NONE)
    private String name;

    @Column(unique = true, nullable = false)
    private String email;

    @Column(nullable = false)
    private String password;
    @JsonIgnore
    @OneToMany(mappedBy = "company", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    List<Coupon> companyCoupons;

    public Company(String name, String email, String password, List<Coupon> companyCoupons) {
        this.name = name;
        this.email = email;
        this.password = password;
        this.companyCoupons = companyCoupons;
    }

    @Override
    public String toString() {
        return "Company{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", email='" + email + '\'' +
                ", password='" + password + " Coupons " + companyCoupons.size();

    }

}

File: coupons3/src/Models/Company.ts
import Coupon from "./Coupon";


class Company {
    public  _id: number;
    public _name: string;
    public _email: string;
    public _password: string;
    public _companyCoupons?: Array<Coupon>;


    constructor(id: number, name: string, email: string, password: string, companyCoupons?: Array<Coupon>) {
        this._id = id;
        this._name = name;
        this._email = email;
        this._password = password;
        this._companyCoupons = companyCoupons;
    }


    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get name(): string {
        return this._name;
    }

    set name(value: string) {
        this._name = value;
    }

    get email(): string {
        return this._email;
    }

    set email(value: string) {
        this._email = value;
    }

    get password(): string {
        return this._password;
    }

    set password(value: string) {
        this._password = value;
    }

    get companyCoupons(): Array<Coupon> {
        return this._companyCoupons;
    }

    set companyCoupons(value: Array<Coupon>) {
        this._companyCoupons = value;
    }

// toJson(): any{
    //     return {
    //         "id": this._id,
    //         "name": this._name,
    //         "email": this._email,
    //         "password": this._password,
    //     }
    // }
}

export default Company